# MICROSOFT MOVIE STUDIO ANALYSIS

## INTRODUCTION

## Business Problem

### Objectives

To solve the above problem and help in steering Microsoft in the right direction in this endeavour, the following are this project's objectives

<b>Main Objective</b>
> Pass

<b>Specific Objectives</b>
1. To ...
2. To ...
3. To ...

## Experimental Design

The research follows the steps as listed:
1. Introduction and Business Problem
2. Loading Data and Relevant Libraries
3. Data Wrangling and Preparation
4. Data Exploration and Visualization
5. Conclusions
6. Recommendations

## Metric of Success
The project will be considered a success once it yields at least three actionable and data-driven recommendations to Mircosoft which can ensure the company focuses on the right steps while breaking into the Movie Industry

## Loading Libraries

In [1]:
import pandas as pd
import numpy as np
import sqlite3
import seaborn as sns
from matplotlib import pyplot as plt
%matplotlib inline

## Loading Data

In [2]:
# Creating a connection to database file
conn = sqlite3.connect("./data/im.db")


# Query for all tables
tablesQuery = """
SELECT name FROM sqlite_master
WHERE type='table';
"""

# Executing query
pd.read_sql(tablesQuery, conn)

name
0   movie_basics
1      directors
2      known_for
3     movie_akas
4  movie_ratings
5        persons
6     principals
7        writers

> From the abov, there are 8 tables, investigating them reveals the following:

### DataBase Files

**Writers Table**

In [3]:
table1 = """
Select * FROM writers
LIMIT 10;
"""
pd.read_sql(table1, conn)

movie_id  person_id
0  tt0285252  nm0899854
1  tt0438973  nm0175726
2  tt0438973  nm1802864
3  tt0462036  nm1940585
4  tt0835418  nm0310087
5  tt0835418  nm0841532
6  tt0878654  nm0284943
7  tt0878654  nm0284943
8  tt0878654  nm0284943
9  tt0996958  nm2286991

> The table above is giving information about the writers responsible for various movies using respective IDs.

**Directors Table**

In [4]:
table2 = """
SELECT * FROM directors
LIMIT 10
"""

pd.read_sql(table2, conn)

movie_id  person_id
0  tt0285252  nm0899854
1  tt0462036  nm1940585
2  tt0835418  nm0151540
3  tt0835418  nm0151540
4  tt0878654  nm0089502
5  tt0878654  nm2291498
6  tt0878654  nm2292011
7  tt0879859  nm2416460
8  tt0996958  nm2286991
9  tt0996958  nm2286991

> The table above is giving information about the directors responsible for various movies using respective IDs.

**Movie Basics Table**

In [5]:
table3 = """
SELECT * FROM movie_basics
LIMIT 10;
"""

pd.read_sql(table3, conn)

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   
5  tt0111414                      A Thin Life                 A Thin Life   
6  tt0112502                          Bigfoot                     Bigfoot   
7  tt0137204                  Joe Finds Grace             Joe Finds Grace   
8  tt0139613                       O Silêncio                  O Silêncio   
9  tt0144449            Nema aviona za Zagreb       Nema aviona za Zagreb   

   start_year  runtime_minutes                      genres  
0        2013            175.0          Action,Crime,Drama  
1        2019            114.0             Biography,Drama  
2        2018            122.0                       Drama  
3        2018              NaN                Comedy,Drama  
4        2017             80.0        Comedy,Drama,Fantasy  
5        2018             75.0                      Comedy  
6        2017              NaN             Horror,Thriller  
7        2017             83.0  Adventure,Animation,Comedy  
8        2012              NaN         Documentary,History  
9        2012             82.0                   Biography

> The above table gives information about basic attributes of movies:
* Titles
* Runtime in Minutes
* Genres spanned
* Start/Release Year

**Movie AKAs**

In [6]:
table4 = """
SELECT * FROM movie_akas
LIMIT 10;
"""
pd.read_sql(table4, conn)

movie_id  ordering                                    title region  \
0  tt0369610        10                            Джурасик свят     BG   
1  tt0369610        11                        Jurashikku warudo     JP   
2  tt0369610        12  Jurassic World: O Mundo dos Dinossauros     BR   
3  tt0369610        13                  O Mundo dos Dinossauros     BR   
4  tt0369610        14                           Jurassic World     FR   
5  tt0369610        15                           Jurassic World     GR   
6  tt0369610        16                           Jurassic World     IT   
7  tt0369610        17                            Jurski svijet     HR   
8  tt0369610        18                             Olam ha'Yura     IL   
9  tt0369610        19           Jurassic World: Mundo Jurásico     MX   

  language        types   attributes  is_original_title  
0       bg         None         None                0.0  
1     None  imdbDisplay         None                0.0  
2     None  imdbDisplay         None                0.0  
3     None         None  short title                0.0  
4     None  imdbDisplay         None                0.0  
5     None  imdbDisplay         None                0.0  
6     None  imdbDisplay         None                0.0  
7     None  imdbDisplay         None                0.0  
8       he  imdbDisplay         None                0.0  
9     None  imdbDisplay         None                0.0

> At first glance, the above table has no relevant information to use in the analysis that is not already in other preceeding tables

**Movie Ratings Table**

In [7]:
table5 = """
SELECT * FROM movie_ratings
LIMIT 10;
"""

pd.read_sql(table5, conn)

movie_id  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21
5   tt1069246            6.2       326
6   tt1094666            7.0      1613
7   tt1130982            6.4       571
8   tt1156528            7.2       265
9   tt1161457            4.2       148

> The above table seems to be conveying the average rating for a movie as well as the number of votes contributing to said rating

**Known For**

In [8]:
table6 = """
SELECT * FROM known_for
LIMIT 10;
"""

pd.read_sql(table6, conn)

person_id   movie_id
0  nm0061671  tt0837562
1  nm0061671  tt2398241
2  nm0061671  tt0844471
3  nm0061671  tt0118553
4  nm0061865  tt0896534
5  nm0061865  tt6791238
6  nm0061865  tt0287072
7  nm0061865  tt1682940
8  nm0062070  tt1470654
9  nm0062070  tt0363631

> Going off on a limb, the data above probably represents actors and which works they are famous for.

**Principals**

In [9]:
table7 = """
SELECT * FROM principals
LIMIT 10;
"""

pd.read_sql(table7, conn)

movie_id  ordering  person_id  category       job  \
0  tt0111414         1  nm0246005     actor      None   
1  tt0111414         2  nm0398271  director      None   
2  tt0111414         3  nm3739909  producer  producer   
3  tt0323808        10  nm0059247    editor      None   
4  tt0323808         1  nm3579312   actress      None   
5  tt0323808         2  nm2694680     actor      None   
6  tt0323808         3  nm0574615     actor      None   
7  tt0323808         4  nm0502652   actress      None   
8  tt0323808         5  nm0362736  director      None   
9  tt0323808         6  nm0811056  producer  producer   

                 characters  
0               ["The Man"]  
1                      None  
2                      None  
3                      None  
4          ["Beth Boothby"]  
5         ["Steve Thomson"]  
6  ["Sir Lachlan Morrison"]  
7   ["Lady Delia Morrison"]  
8                      None  
9                      None

> The above data will not be used in the analysis to follow.

**Persons**

In [10]:
table8 = """
SELECT * FROM persons
LIMIT 10;
"""

pd.read_sql(table8, conn)

person_id       primary_name  birth_year death_year  \
0  nm0061671  Mary Ellen Bauder         NaN       None   
1  nm0061865       Joseph Bauer         NaN       None   
2  nm0062070         Bruce Baum         NaN       None   
3  nm0062195       Axel Baumann         NaN       None   
4  nm0062798        Pete Baxter         NaN       None   
5  nm0062879     Ruel S. Bayani         NaN       None   
6  nm0063198              Bayou         NaN       None   
7  nm0063432      Stevie Be-Zet         NaN       None   
8  nm0063618          Jeff Beal      1963.0       None   
9  nm0063750    Lindsay Beamish         NaN       None   

                                 primary_profession  
0         miscellaneous,production_manager,producer  
1        composer,music_department,sound_department  
2                        miscellaneous,actor,writer  
3  camera_department,cinematographer,art_department  
4  production_designer,art_department,set_decorator  
5         director,production_manager,miscellaneous  
6                                             actor  
7                               composer,soundtrack  
8              composer,music_department,soundtrack  
9                             actress,miscellaneous

> This table details what professions key individuals in the database do and when they were born/laid to rest; it is not relevant to the ensuing analysis.

## Loading Data

### From Database

In [24]:
# Joining relevant information into a single table
movieBasicsDf = pd.read_sql("""
                SELECT primary_title, start_year, runtime_minutes, genres, averagerating, numvotes
                FROM movie_basics
                JOIN movie_ratings
                    USING(movie_id)
                WHERE genres IS NOT NULL;
                """, conn)

print(movieBasicsDf.shape, end="\n\n")
movieBasicsDf.head()

(73052, 6)



primary_title  start_year  runtime_minutes  \
0                        Sunghursh        2013            175.0   
1  One Day Before the Rainy Season        2019            114.0   
2       The Other Side of the Wind        2018            122.0   
3                  Sabse Bada Sukh        2018              NaN   
4         The Wandering Soap Opera        2017             80.0   

                 genres  averagerating  numvotes  
0    Action,Crime,Drama            7.0        77  
1       Biography,Drama            7.2        43  
2                 Drama            6.9      4517  
3          Comedy,Drama            6.1        13  
4  Comedy,Drama,Fantasy            6.5       119

## From Flat Files

In [14]:
!ls ./data

bom.movie_gross.csv
im.db
rt.movie_info.tsv
rt.reviews.tsv
tmdb.movies.csv
tn.movie_budgets.csv


In [22]:
movieGrossDf = pd.read_csv("./data/bom.movie_gross.csv")
movieInfoDf = pd.read_csv("./data/rt.movie_info.tsv", delimiter="\t")
movieBudgetsDf = pd.read_csv("./data/tn.movie_budgets.csv")
imdbRatingsDf = pd.read_csv("./data/tmdb.movies.csv", index_col=0)

### Basic Information from Datasets

In [25]:
# Movie Basics
movieBasicsDf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73052 entries, 0 to 73051
Data columns (total 6 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   primary_title    73052 non-null  object 
 1   start_year       73052 non-null  int64  
 2   runtime_minutes  65720 non-null  float64
 3   genres           73052 non-null  object 
 4   averagerating    73052 non-null  float64
 5   numvotes         73052 non-null  int64  
dtypes: float64(2), int64(2), object(2)
memory usage: 3.3+ MB


> The above data has 6 columns and 73,052 rows.
All of the null values are focused around runtime length.
The data types for respective fields is seemingly appropriate. 

In [26]:
# Movie Gross
movieGrossDf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


> The above data has 3387 rows and 5 columns.

1. Studio field has 5 null values
2. Domestic gross has 28 null values
3. Foreign gross has 350 null values

The only field with an incorrect data type for analysis purpose is Foreign Gross. This needs to be converted to numeric.

In [27]:
# Movie Info
movieInfoDf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            1560 non-null   int64 
 1   synopsis      1498 non-null   object
 2   rating        1557 non-null   object
 3   genre         1552 non-null   object
 4   director      1361 non-null   object
 5   writer        1111 non-null   object
 6   theater_date  1201 non-null   object
 7   dvd_date      1201 non-null   object
 8   currency      340 non-null    object
 9   box_office    340 non-null    object
 10  runtime       1530 non-null   object
 11  studio        494 non-null    object
dtypes: int64(1), object(11)
memory usage: 146.4+ KB


In [28]:
movieInfoDf.head()

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

> The data has 12 columns and 1560 rows with a lot of null values

The columns with meaningul information to the analysis - rating, director, runtime and studio, need wrangling(Consistency and Validity concerns) to be analysis ready.

In [29]:
# Movie Budgets

movieBudgetsDf.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


> The above data has 6 columns and 5782 rows with numbers-focused fields being casted as object-types.
There are no missing values.

In [30]:
# IMDB data

imdbRatingsDf.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 26517 entries, 0 to 26516
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   genre_ids          26517 non-null  object 
 1   id                 26517 non-null  int64  
 2   original_language  26517 non-null  object 
 3   original_title     26517 non-null  object 
 4   popularity         26517 non-null  float64
 5   release_date       26517 non-null  object 
 6   title              26517 non-null  object 
 7   vote_average       26517 non-null  float64
 8   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(2), object(5)
memory usage: 2.0+ MB


> The data above has 9 columns and 26517 rows with no missing values.
Release date needs to be cast as datetime

## Data Cleaning

## Movie Basics Dataset

In [31]:
# Preview the first five rows

movieBasicsDf.head()

primary_title  start_year  runtime_minutes  \
0                        Sunghursh        2013            175.0   
1  One Day Before the Rainy Season        2019            114.0   
2       The Other Side of the Wind        2018            122.0   
3                  Sabse Bada Sukh        2018              NaN   
4         The Wandering Soap Opera        2017             80.0   

                 genres  averagerating  numvotes  
0    Action,Crime,Drama            7.0        77  
1       Biography,Drama            7.2        43  
2                 Drama            6.9      4517  
3          Comedy,Drama            6.1        13  
4  Comedy,Drama,Fantasy            6.5       119